includes/class-wcs-cart-renewal.php

```diff
@@ -113,6 +113,7 @@ public function setup_hooks() {
 		// Make sure renewal meta data persists between sessions
 		add_filter( 'woocommerce_get_cart_item_from_session', array( &$this, 'get_cart_item_from_session' ), 10, 3 );
 		add_action( 'woocommerce_cart_loaded_from_session', array( &$this, 'cart_items_loaded_from_session' ), 10 );
+		add_action( 'woocommerce_cart_loaded_from_session', array( &$this, 'restore_order_awaiting_payment' ), 10, 1 );
```

Comment:
Not essential, but a couple of tweaks we could consider:

1. Removing `&` from `&$this` → I guess this is in sympathy with/in keeping with the style of the preceding lines, but we're already working with a reference.
2. No need to specify the number of accepted args, since we're using the default (though, I'd also be interested if there are reasons for being explicit about this).

```suggestion
		add_action( 'woocommerce_cart_loaded_from_session', array( $this, 'restore_order_awaiting_payment' ), 10 );
```